src/codegen/function_body_chunk.rs

```diff
@@ -718,13 +718,16 @@ impl Builder {
                 type_: "*mut _".to_string(),
             });
         }
-        finish_args.push(Chunk::Name("res".to_string()));
         finish_args.extend(
             trampoline
                 .output_params
                 .iter()
-                .filter(|param| param.direction == ParameterDirection::Out)
+                .filter(|param| param.direction == ParameterDirection::Out ||
+                                param.typ.full_name(&env.library) == "Gio.AsyncResult")
                 .map(|param| {
+                    if param.typ.full_name(&env.library) == "Gio.AsyncResult" {
+                        return Chunk::Name("res".to_string());
```

Comment:
Should panic/error out if there is no such parameter. Otherwise looks like this should work.

Reply:
At this point, this is already way too late. I *think* it's already checked beforehand.

Reply:
Good point! If we ever panic because of it, a fix will be required. I'll add that in the assert message. :D